cmd/access/access_node_builder.go

```diff
@@ -0,0 +1,200 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/cmd"
+	"github.com/onflow/flow-go/cmd/build"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/local"
+	"github.com/onflow/flow-go/network"
+	jsoncodec "github.com/onflow/flow-go/network/codec/json"
+	"github.com/onflow/flow-go/network/p2p"
+	"github.com/onflow/flow-go/network/topology"
+	"github.com/onflow/flow-go/network/validator"
+)
+
+// AccessNodeBuilder is initializes and runs an Access node either as a staked node or an unstaked node
+// It is composed of the FlowNodeBuilder
+type AccessNodeBuilder struct {
+	*cmd.FlowNodeBuilder
+	staked                  bool
+	stakedAccessNodeIDHex   string
+	unstakedNetworkBindAddr string
+	unstakedNetwork         *p2p.Network
+	unstakedMiddleware      *p2p.Middleware
+}
+
+func FlowAccessNode() *AccessNodeBuilder {
```

Comment:
We could move this and the existing `FlowNodeBuilder` to a `scaffold` package together to avoid all the newly public methods in `FlowNodeBuilder` and keep the scaffolding logic together.

Reply:
Lemme do this in the other [PR](https://github.com/onflow/flow-go/pull/983) since that one is primarily for such scaffold related changes.

Reply:
Actually, lemme do that as different PR all together once I have [983](https://github.com/onflow/flow-go/pull/983) merged into this one otherwise it will just become even more difficult to review.

Reply:
created [this](https://github.com/dapperlabs/flow-go/issues/5688) issue to address this comment